### PR TITLE
Cache the http client in order to reuse the connection

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Fix: #27726

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/cmd/pilot-agent/status/server.go [hunk 1]
```diff
@@ -169,6 +172,15 @@ func NewServer(config Config) (*Server, error) {
 		if prober.HTTPGet.Port.Type != intstr.Int {
 			return nil, fmt.Errorf("invalid prober config for %v, the port must be int type", path)
 		}
+		// Construct a http client and cache it in order to reuse the connection.
+		s.appProbeClient[path] = &http.Client{
```
Comment (bianpengyuan) [inline pilot/cmd/pilot-agent/status/server.go:176]: Why not use the same client for all paths? The client seem to be equivalent.

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/status/server.go:176]: As you can see below, the timeout maybe different

Comment (bianpengyuan) [reply pilot/cmd/pilot-agent/status/server.go:176]: I see. thanks!

**Review by ramaraochavali — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**